drill-admin/src/main/kotlin/com/epam/drill/endpoints/agent/DrillServerWs.kt

```diff
@@ -56,38 +56,37 @@ class DrillServerWs(override val kodein: Kodein) : KodeinAware {
         }
     }
 
-    @UseExperimental(ImplicitReflectionSerializer::class)
     private suspend fun subscribe(
         wsSession: DrillWsSession,
         event: Message
     ) {
         sessionStorage += (wsSession)
         println("${event.destination} is subscribed")
+        sendToAllSubscribed(event.destination)
+    }
+
+    @UseExperimental(ImplicitReflectionSerializer::class)
+    suspend fun sendToAllSubscribed(destination: String) {
         app.run {
             wsTopic {
-                val resolve = resolve(event.destination, sessionStorage)!!
-                if (resolve is Collection<*>) {
-                    sessionStorage.sendTo(
-                        Message(
-                            MessageType.MESSAGE,
-                            event.destination,
-                            Json.stringify(resolve)
-                        )
-                    )
-                } else {
-                    @Suppress("UNCHECKED_CAST") val serializer = resolve::class.serializer() as KSerializer<Any>
-                    sessionStorage.sendTo(
-                        Message(
-                            MessageType.MESSAGE,
-                            event.destination,
-                            serializer stringify resolve
-                        )
-                    )
+                val resolve = resolve(destination, sessionStorage)!!
+                val message = when (resolve) {
+                    is Collection<*> -> Json.stringify(resolve)
```

Comment:
expand this!
try to rewrite this part of bf to **when** operator.
when( List) -> serializer.list
when(Set) -> serializer.set
etc..
Remove autogen Json